src/Compilers/Core/Portable/SourceGeneration/IncrementalGeneratorInitializationContext_ForAttributeWithSimpleName.cs

```diff
@@ -0,0 +1,284 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Transactions;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.SourceGeneration;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis;
+
+using Aliases = ArrayBuilder<(string aliasName, string symbolName)>;
+
+public partial struct IncrementalGeneratorInitializationContext
+{
+    private static readonly ObjectPool<Stack<string>> s_stackPool = new(static () => new());
+
+    /// <summary>
+    /// Returns all syntax nodes of that match <paramref name="predicate"/> if that node has an attribute on it that
+    /// could possibly bind to the provided <paramref name="simpleName"/>. <paramref name="simpleName"/> should be the
+    /// simple, non-qualified, name of the attribute, including the <c>Attribute</c> suffix, and not containing any
+    /// generics, containing types, or namespaces.  For example <c>CLSCompliantAttribute</c> for <see
+    /// cref="System.CLSCompliantAttribute"/>.
+    /// <para/> This provider understands <see langword="using"/> (<c>Import</c> in Visual Basic) aliases and will find
+    /// matches even when the attribute references an alias name.  For example, given:
+    /// <code>
+    /// using XAttribute = System.CLSCompliantAttribute;
+    /// [X]
+    /// class C { }
+    /// </code>
+    /// Then
+    /// <c>context.SyntaxProvider.CreateSyntaxProviderForAttribute(nameof(CLSCompliantAttribute), (node, c) => node is ClassDeclarationSyntax)</c>
+    /// will find the <c>C</c> class.
+    /// </summary>
+    internal IncrementalValuesProvider<SyntaxNode> ForAttributeWithSimpleName(
+        string simpleName,
+        Func<SyntaxNode, CancellationToken, bool> predicate)
+    {
+        var syntaxHelper = this.SyntaxHelper;
+
+        // Create a provider that provides (and updates) the global aliases for any particular file when it is edited.
+        var individualFileGlobalAliasesProvider = this.SyntaxProvider.CreateSyntaxProvider(
```

Comment:
Again I think this can be optimally implemented via a syntax strategy, but I'm ok taking as-is for now and replacing the internals later.

Reply:
This is based off of a pretty old version of this branch, but was where I started in converting it for reference: https://github.com/chsienki/roslyn/commit/b87523e82632ed1b43d41c4ccdefbb1ecb8d1467

Reply:
i don't really understand what benefit is gotten from this "syntax stragy" approach.  WHat is the net benefit to the user at end of the day?

Reply:
I think that is what @chsienki was saying here. Essentially we can make this even better by getting rid of the `SemanticModel` dependency in the future. I'm several layers of speculation into his comment though so will probably just wait for him to actually tell us what he thought when he gets back. 😄 

I don't think it's a blocker though. It's obviously a huge improvement over where we are now but it also seems like we can make it even better with some more work.

Reply:
After thinking deeply on this, for all of 5 minutes, I think that we should defer this to a second PR. It's essentially a new feature in generators, ability to have a pure syntax consumption story. That would make sense as a second PR where we implement the feature and flip this API over to using it and validate the tests all flip as we expect. 

But will wait and see if @chsienki agrees when he's back.

Reply:
Basically what @jaredpar said, its possible to make this _more_ efficient if we re-write it as an `ISyntaxSelectionStrategy<T>` but it won't affect the public API and we should not block on it. Lets do it in a follow up PR at some point in the future.